update.ps1

```diff
@@ -65,145 +50,191 @@ function Get-ADPAccessToken {
-            } | ConvertTo-Json -Depth 10
-
-            $splatParams = @{
-                Uri         = "$($config.BaseUrl)/events/hr/v1/worker.business-communication.email.change"
-                Method      = 'POST'
-                Body        = $body
-                Headers     = $headers
-                Certificate = $certificate
-                ContentType = 'application/json'
-            }
-            if (-not($dryRun -eq $true)) {
-                $responseUpdateUser = Invoke-RestMethod @splatParams
-                if ($responseUpdateUser.events[0].eventStatusCode.codeValue -eq 'submitted') {
-                    $accountReference = $responseGetUser.Workers[0].associateOID
-                    $success = $true
-                    $auditLogs.Add([PSCustomObject]@{
-                            Message = "Updated E-mail address for: $($p.DisplayName) from: [$($responseGetUser.Workers[0].businessCommunication.emails[0].emailUri)] to: [$($account.workerEmail)]"
-                            IsError = $false
                         })
+                } | ConvertTo-Json -Depth 10
+    
+                $splatParams = @{
+                    Uri         = "$($actionContext.Configuration.BaseUrl)/events/hr/v1/worker.business-communication.email.change"
+                    Method      = 'POST'
+                    Body        = $body
+                    Headers     = $headers
+                    Certificate = $certificate
+                    ContentType = 'application/json'
+                }
+                if (-not($dryRun -eq $true)) {
+                    $responseUpdateUser = Invoke-RestMethod @splatParams
+                    if ($responseUpdateUser.events[0].eventStatusCode.codeValue -eq 'submitted') {
+                        $outputContext.AccountReference = $responseGetUser.Workers[0].associateOID
+                        $outputContext.Success = $true
+                        $outputContext.AuditLogs.Add([PSCustomObject]@{
+                                Message = "Updated E-mail address for: $($personContext.Person.DisplayName) from: [$($responseGetUser.Workers[0].businessCommunication.emails[0].emailUri)] to: [$($actionContext.Data.workerEmail)]"
+                                IsError = $false
+                            })
+                    }
                 }
             }
-        }
+    
+            'NoChanges' {
+                $outputContext.Success = $true
+                $outputContext.AuditLogs.Add([PSCustomObject]@{
+                        Message = "No changes needed for account $($personContext.Person.DisplayName)"
+                        IsError = $false
+                    })
+                break
+            }
 
-        'Exit' {
-            $success = $true
-            $auditLogs.Add([PSCustomObject]@{
-                    Message = $msg
-                    IsError = $false
-                })
-            break
+            'NotFound' {
+                $outputContext.Success = $true
```

Comment:
I think we should change this to `$false` because, at this point, HelloID expected an account to be present in ADP Workforce.